### ISSUE TITLE
Self not detected as not defined for node env

### ISSUE THREAD

**Issue by BloodyStupidViktor** [eslint/eslint#7007]
Self not detected as not defined, I guess similar to issue #2151.
I only have the node environment enabled. When I remove node from env, self is correctly highlighted as undefined. I guess I can keep it that way and copy from the globals file.
However, having a look at it, the node environment does not declare a self global. The only environments that do are browser, worker and serviceworker, which I don't have.
Any ideas why this is happening?
Using eslint@2.13.1, which uses globals@9.9.0.

P.S. Can anyone update me on the ability to have different environments / globals in different subfolders? Read a couple of issues and skimmed over the changelog, but couldn't find much definitive information.

P.P.S. Thanks to all contributors for all the effort!

**Comment by eslintbot** [comment]
Hi @BloodyStupidViktor, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by vitorbal** [comment]
Hi @BloodyStupidViktor, thanks for the report. Besides what the bot mentions above, could you also verify if the problem persists in the latest version of ESLint? Thanks!

**Comment by kaicataldo** [comment]
I just tried this with the latest release of ESLint and am not seeing this behavior.

Linting
```js
module.exports = function() {};

var test = self;
```
with the following config
```json
{
  "env": {
    "node": true
  },
  "rules": {
    "no-undef": "error"
  }
}
```
yields this error:
```
  3:10  error  'self' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)

npm ERR! Test failed.  See above for more details.
```

**Comment by kaicataldo** [comment]
>P.S. Can anyone update me on the ability to have different environments / globals in different subfolders? Read a couple of issues and skimmed over the changelog, but couldn't find much definitive information.

Check out this section of the configuration guide: http://eslint.org/docs/user-guide/configuring#extending-configuration-files

For specific questions and help, please feel free to drop by our [Gitter](https://gitter.im/eslint/eslint).

**Comment by eslintbot** [comment]
Hi @BloodyStupidViktor, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by kaicataldo** [comment]
Sorry for all the label switching - I goofed. @BloodyStupidViktor we'll be able to help you a lot better if you can please fill out the issue template :)

**Comment by vitorbal** [comment]
friendly ping @BloodyStupidViktor, could you take a look at our questions above? Otherwise, I'm afraid we don't have enough info to help you. Thanks!

**Comment by vitorbal** [comment]
Still no response, so closing.
